kernel/CMakeLists.txt

```diff
@@ -52,7 +52,10 @@ target_sources_ifdef(
   userspace.c
   )
 
+get_property(E_KERNEL_ENTRY GLOBAL PROPERTY E_KERNEL_ENTRY)
 
 add_dependencies(kernel offsets_h)
 
-target_link_libraries(kernel zephyr_interface)
+target_link_libraries(kernel PUBLIC zephyr_interface)
+target_link_libraries(kernel INTERFACE ${E_KERNEL_ENTRY} ${TOOLCHAIN_LIBS})
```

Comment:
Seems to me that TOOLCHAIN_LIBS is not specific to the kernel, but rather something that all Zephyr sources need. So it should be

```
sebo@mach:~/zephyr$ sebo@mach:~/zephyr$ git diff | cat 
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b66239f70..166075e087 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -693,6 +693,8 @@ set_property(TARGET
 
 get_property(E_KERNEL_ENTRY GLOBAL PROPERTY E_KERNEL_ENTRY)
 
+zephyr_link_libraries(${TOOLCHAIN_LIBS})
+
 set(zephyr_lnk
   ${LINKERFLAGPREFIX},-Map=${PROJECT_BINARY_DIR}/${KERNEL_MAP_NAME}
   ${LINKERFLAGPREFIX},--start-group
diff --git a/kernel/CMakeLists.txt b/kernel/CMakeLists.txt
index 04111eaf6e..ae6de19ff6 100644
--- a/kernel/CMakeLists.txt
+++ b/kernel/CMakeLists.txt
@@ -57,5 +57,5 @@ get_property(E_KERNEL_ENTRY GLOBAL PROPERTY E_KERNEL_ENTRY)
 add_dependencies(kernel offsets_h)
 
 target_link_libraries(kernel PUBLIC zephyr_interface)
-target_link_libraries(kernel INTERFACE ${E_KERNEL_ENTRY} ${TOOLCHAIN_LIBS})
+target_link_libraries(kernel INTERFACE ${E_KERNEL_ENTRY})
 

```

Reply:
Point taken.
Smaller commits but single PR.

Reply:
I added the TOOLCHAIN_LIBS to the kernel, as it appeared to me this was the lowest part of the Zephyr system which requires the TOOLCHAIN_LIBS.

From there, the TOOLCHAIN_LIBS is propagated to other who depends on kernel, as does libzephyr.a, and the rest depends on zephyr.

Normally I would have placed such flag on the CMAKE_EXE_LINKER_FLAGS, but this is not used with Zephyr, thus I found kernel linking dependency to be the second best option.
Of course this is open to other opinions.

Please note, the flag is not required for compiling, only linking.
So the all Zephyr sources doesn't need it.

Reply:
This use-case is covered by zephyr_link_libraries().

So you either have to use zephyr_link_libaries, or block this PR on another PR that deprecates
zephyr_link_libraries().

Reply:
Well in that case, this zephyr_interface  description should be
updated, as this is a link flag, which much be used by the
executable that links and not  directly matching the description in
CMakeLists.txt today about zephyr_interface.
# zephyr_interface is a source-less library that has all the global                                     
# compiler options needed by all source files. All zephyr libraries,                                    
# including the library named "zephyr" link with this library to                                        
# obtain these flags.

I can update the description and move the link flag here instead.